src/Handlers/PaginatedPageHandler.php

```diff
@@ -46,7 +46,11 @@ public function handle($file, PageData $pageData)
         return $this->paginator->paginate(
             $file,
             $pageData->get($pageData->page->pagination->collection),
-            $pageData->page->pagination->perPage ?: ($pageData->page->perPage ?: 10)
+            $pageData->page->pagination->perPage ?: (
+                $pageData->page->collections->{$pageData->page->pagination->collection}->perPage ?: (
+                    $pageData->page->perPage ?: 10
+                )
+            )
```

Comment:
I think it would be a little cleaner to use a couple of temp variables, and then a chained ternary, here:

```php
$page = $pageData->page;
$collection = $page->pagination->collection;
$page->addVariables($this->getPageVariables($file));

return $this->paginator->paginate(
    $file,
    $pageData->get($collection),
    $page->pagination->perPage 
        ?: $page->collections->{$collection}->perPage 
        ?: $page->perPage
        ?: 10
)->map ...
```

Reply:
Agreed, that's a lot cleaner. I had to swap around the order of the temporary variables to get it to work, but that's in place now.